src/basic/io-util.c

```diff
@@ -125,21 +128,39 @@ int loop_write(int fd, const void *buf, size_t nbytes, bool do_poll) {
                 p = buf;
         }
 
+        end = timestamp_is_set(timeout) ? usec_add(now(CLOCK_MONOTONIC), timeout) : USEC_INFINITY;
+
         do {
                 ssize_t k;
+                usec_t t;
+
+                if (end != USEC_INFINITY) {
+                        t = now(CLOCK_MONOTONIC);
+                        if (t >= end)
+                                return -ETIME;
+                }
```

Comment:
hmm, so i think we should try to write first, and only check the timeout after we couldn't write right away

Reply:
Hmm, I can't seem to imagine how to make things work like this? Also this is "cheating" in my opinion, and a casual reader could be confused by the fact that the timeout doesn't always apply.